docs/api/webviews.md

```diff
@@ -0,0 +1,355 @@
+```
+
+### `replaceText(text)`
+
+Replace the text of the element with the specified text.
+
+:::note
+This action is currently not supported for content-editable elements.
+:::
+
+```js
+await web.element(by.web.id('identifier')).replaceText('Hello World!');
+```
+
+### `clearText()`
+
+Clear the text of the element.
+
+:::note
+This action is currently not supported for content-editable elements.
+:::
+
+```js
+await web.element(by.web.id('identifier')).clearText();
+```
+
+### `selectAllText()`
+
+Select all the text of the element.
+
+:::note
+This action is currently supported for content-editable elements only.
+:::
+
+```js
+await web.element(by.web.id('identifier')).selectAllText();
+```
+
+### `getText()`
+
+Get the text of the element.
+
+```js
+const text = await web.element(by.web.id('identifier')).getText();
+```
+
+### `scrollToView()`
+
+Scroll to the element until its top is at the top of the viewport.
+
+```js
+await web.element(by.web.id('identifier')).scrollToView();
+```
+
+### `focus()`
+
+Focus on the element.
+
+```js
+await web.element(by.web.id('identifier')).focus();
+```
+
+### `moveCursorToEnd()`
+
+Move the input cursor to the end of the element's content.
+
+:::note
+This action is currently supported for content-editable elements only.
+:::
+
+```js
+await web.element(by.web.id('identifier')).moveCursorToEnd();
+```
+
+### `runScript(script)`
+
+Run the specified script on the element.
+
+The script should be a string that contains a valid JavaScript function.
+It will be called with that element as the first argument.
+
+```js
+const webElement = web.element(by.web.id('identifier'));
+await webElement.runScript(`function foo(element) {
+  console.log(element);
+}`);
+```
+
+### `runScriptWithArgs(script, ...args)`
+
+Run the specified script on the element with extra arguments.
+
+The script should be a string that contains a valid JavaScript function.
+It will be called with the specified arguments and the element itself as the last argument.
+
+```js
+const webElement = web.element(by.web.id('identifier'));
+await webElement.runScriptWithArgs(`function foo(arg1, arg2, element) {
+  console.log(arg1, arg2, element);
+}`, "foo", 123);
+```
+
+### `getCurrentUrl()`
+
+Get the current URL of the web view.
+
+:::note
+Although this action returns the URL of the presented web document, it can be called from an inner element only (for example, an iframe id or the HTML) and not from the root native web view element itself.
+:::
+
+```js
+const url = await web.element(by.web.id('identifier')).getCurrentUrl();
+```
+
+### `getTitle()`
+
+Get the title of the web view.
+
+:::note
+Although this action returns the title of the presented web document, it can be called from an inner element only (for example, an iframe id or the HTML) and not from the root native web view element itself.
+:::
+
+```js
+const title = await web.element(by.web.id('identifier')).getTitle();
+```
```

Comment:
I really don't like these APIs. It can be useful when testing an inner **iframe** URL or title, but not convenient when trying to test the root document. It will probably work when targeting the HTML tag element using `by.web.tag("html")` but I haven't tried that yet.

We should expose the same APIs for the web-view element (the root native element), for running the same operations on the web-view element (the main frame).

The Espresso code we're running with this APIs is:
```java
return getWebViewInteraction().withElement(get()).perform(Atoms.getCurrentUrl()).get();
```
https://github.com/wix/Detox/blob/master/detox/android/detox/src/full/java/com/wix/detox/espresso/web/WebElement.java#L65-L79 (`WebElement`)

While we should enable operating on the root frame **without specifying the inner element** (from `WebViewElement` object) is something like:
```java
return getWebViewInteraction().reset().perform(Atoms.getCurrentUrl()).get();
```

Should be in the same API level with `web.element(..)` https://github.com/wix/Detox/blob/master/detox/android/detox/src/full/java/com/wix/detox/espresso/web/WebViewElement.java

Reply:
I've opened a new issue for this: https://github.com/wix/Detox/issues/3752